library/src/engine/engine.ts

```diff
@@ -875,15 +875,15 @@ function applyAttributePlugin(
   attrKey: string,
   value: string,
 ): void {
-  const rawKey = camel(alias ? attrKey.slice(alias.length) : attrKey)
```

Comment:
This was calling camel just to lowercase the first letter. Calling camel messed things up when there were literal dashes in the key though.